baseplate/clients/redis_cluster.py

```diff
@@ -501,9 +527,7 @@ def execute(self, **kwargs: Any) -> Any:
             start_time = perf_counter()
             labels = {
                 f"{PROM_LABELS_PREFIX}_command": "pipeline",
-                f"{PROM_LABELS_PREFIX}_client_name": self.connection_pool.connection_kwargs.get(
-                    "client_name", ""
-                ),
+                f"{PROM_LABELS_PREFIX}_client_name": self.redis_client_name,
```

Comment:
Thanks! As far as I know, the connection pool does have a client name as well but I don't think anyone ever set it. This would technically also be a backwards incompatible change ([example](https://github.com/redis/redis-py/blob/master/tests/test_connection_pool.py#L82) from redis-py tests)

Reply:
We are trying to avoid setting `client_name` on the connection pool as this causes the issue described in the PR description. That is why we always default the value of  `redis_client_name` to the value of the same value as `connection.client_name`.